readme.md
# Stats

```
heat-pipe                                mk1         mk2
-------------------------------------------------------------------------
max_health                               200         300
heat_buffer.max_temperature              1000        2000
heat_buffer.max_transfer                 1GW         8GW
heat_buffer.minimum_glow_temperature     350         700


steam-engine                             mk1         mk2         mk3
-------------------------------------------------------------------------
max_health                               400         600         1200
effectivity                              1           1.5         2
fluid_usage_per_tick                     0.5         0.75        1


steam-turbine                            mk1         mk2         mk3
-------------------------------------------------------------------------
max_health                               300         600         900
effectivity                              1           1.25        1.5
fluid_usage_per_tick                     1           1.5         2
maximum_temperature                      500         700         900


nuclear-reactor                          mk1         mk2         mk3
-------------------------------------------------------------------------
max_health                               500         1000        1500
energy_source.effectivity                1           1.5         2
maximum_temperature                      500         750         1000
neighbour_bonus                          1           1.1         1.2
consumption                              40MW        80MW        160MW
heat_buffer.maximum_temperature          1000        1500        2000
heat_buffer.max_transfer                 10GW        20GW        40GW
heat_buffer.minimum_glow_temperature     350         525         700
heat_buffer.specific_heat                10MJ        20MJ        40MJ


boiler                                   mk1         mk2         mk3
-------------------------------------------------------------------------
max_health                               200         400         600
energy_consumption                       1.8MW       3.6MW       7.2MW
energy_source.emissions_per_minute       30          25          20


heat-exchanger                           mk1         mk2         mk3
-------------------------------------------------------------------------
max_health                               200         400         600
minimum_glow_temperature                 350         450         550
energy_consumption                       10MW        20MW        30MW
target_temperature                       500         600         700
```


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.4.3
Date: 20. 02. 2020
  Changes :
    - Added Mk2/3 (Heat Exchangers / Steam Turbines / Nuclear Reactors)
    - Added Mk2 Heat Pipes
    - Added research Nuclear-2 / Nuclear-3
---------------------------------------------------------------------------------------------------
Version: 0.4.2
Date: 15. 02. 2020
  Fixes :
    - Fixes substation mk3 wire length and supply area
    - Fixes normal graphics with mk2/mk3 big power poles
---------------------------------------------------------------------------------------------------
Version: 0.4.1
Date: 31. 01. 2020
  Changes :
    - General code refactor
    - If a mod has set fast_replaceable_group for a technology family lets use it too
  Fixes :
    - Cleared icon_mipmaps warnings, until FE+ graphics are updated
---------------------------------------------------------------------------------------------------
Version: 0.4.0
Date: 21. 01. 2020
  Changes :
    - Support 0.18
---------------------------------------------------------------------------------------------------
Version: 0.3.0
Date: 30. 12. 2019
  Changes :
    - Added thumbnail
    - Reordered changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.2.5
Date: 09. 07. 2019
  Changes :
    - Removing loop for setting accumulators collision box and fast replacable group.
    - Breaks being able to use upgrade planner to go from vanilla accumulators to FE+ accumulators
---------------------------------------------------------------------------------------------------
Version: 0.2.4
Date: 07. 01. 2019
  Changes :
    - Added check for bobpower and not update vanilla solar panel collision if installed, this mainly because of 0.17.53 update
---------------------------------------------------------------------------------------------------
Version: 0.2.3
Date: 03. 29. 2019
  Changes :
    - Renamed "electric-energy-accumulators-1" to "electric-energy-accumulators" because factorio 0.17.23 changed this
---------------------------------------------------------------------------------------------------
Version: 0.2.2
Date: 03. 10. 2019
  Changes :
    - Changing effectivity to 1 to match base game
---------------------------------------------------------------------------------------------------
Version: 0.2.1
Date: 03. 03. 2019
  Changes :
    - Updating graphics and properties
---------------------------------------------------------------------------------------------------
Version: 0.2.0
Date: 03. 01. 2019
  Changes :
    - Updating to work with 0.17
---------------------------------------------------------------------------------------------------
Version: 0.1.3
Date: 03. 30. 2018
  Changes :
    - Re-balancing science


prototypes/entity/steam-engine.lua
-- steam-engine                             mk1         mk2         mk3
-- max_health                               400         600         800
-- effectivity                              1           1.5         2
-- fluid_usage_per_tick                     0.5         0.75        1
--
-- If someone has already set the fast_replaceable_group lets use it too
if not data.raw["generator"]["steam-engine"].fast_replaceable_group then
    data.raw["generator"]["steam-engine"].fast_replaceable_group = "steam-engine"
end
data.raw["generator"]["steam-engine"].next_upgrade = "steam-engine-mk2"

local mk2 = table.deepcopy(data.raw["generator"]["steam-engine"])
mk2.name = "steam-engine-mk2"
mk2.icon = "__FactorioExtended-Plus-Power__/graphics/icons/" .. mk2.name .. ".png"
mk2.icon_size = 32
mk2.icon_mipmaps = nil
mk2.minable.result = mk2.name
mk2.max_health = 600
mk2.next_upgrade = "steam-engine-mk3"
mk2.effectivity = 1.5
mk2.fluid_usage_per_tick = 0.75

mk2.horizontal_animation.layers[1].filename = "__FactorioExtended-Plus-Power__/graphics/entity/" .. mk2.name .. "/steam-engine-H.png"
mk2.horizontal_animation.layers[1].hr_version.filename = "__FactorioExtended-Plus-Power__/graphics/entity/" .. mk2.name .. "/hr-steam-engine-H.png"
mk2.vertical_animation.layers[1].filename = "__FactorioExtended-Plus-Power__/graphics/entity/" .. mk2.name .. "/steam-engine-V.png"
mk2.vertical_animation.layers[1].hr_version.filename = "__FactorioExtended-Plus-Power__/graphics/entity/" .. mk2.name .. "/hr-steam-engine-V.png"

local mk3 = table.deepcopy(data.raw["generator"]["steam-engine"])
mk3.name = "steam-engine-mk3"
mk3.icon = "__FactorioExtended-Plus-Power__/graphics/icons/" .. mk3.name .. ".png"
mk3.icon_size = 32
mk3.icon_mipmaps = nil
mk3.minable.result = mk3.name
mk3.max_health = 800
mk3.next_upgrade = nil
mk3.effectivity = 2
mk3.fluid_usage_per_tick = 1

mk3.horizontal_animation.layers[1].filename = "__FactorioExtended-Plus-Power__/graphics/entity/" .. mk3.name .. "/steam-engine-H.png"
mk3.horizontal_animation.layers[1].hr_version.filename = "__FactorioExtended-Plus-Power__/graphics/entity/" .. mk3.name .. "/hr-steam-engine-H.png"
mk3.vertical_animation.layers[1].filename = "__FactorioExtended-Plus-Power__/graphics/entity/" .. mk3.name .. "/steam-engine-V.png"
mk3.vertical_animation.layers[1].hr_version.filename = "__FactorioExtended-Plus-Power__/graphics/entity/" .. mk3.name .. "/hr-steam-engine-V.png"

data:extend({mk2, mk3})
